semantics/src/traversal/expressions.rs

```diff
@@ -157,10 +157,12 @@ fn expr_subscript(
                     // Index value is ignored. We may want to introduce a new location
                     // variant named StorageRuntime or something to suit this case.
                     Type::Map(_) => Location::Storage { index: 0 },
+                    Type::Tuple(_) => Location::Value,
```

Comment:
should be memory

Reply:
Assuming that each Tuple is always a fixed size set of base types, why wouldn't it be `Location::Value`? I'm not arguing against it, there's probably a good reason for that, that I want to learn :)

Reply:
I'm thinking that a tuple could also contain arrays or other tuples, in addition to base types.

Even if we did just allow base types in tuples, though, I'd think we'd still need to store them in memory, as I'm not sure how we would manage storing multiple values associated with a tuple on the stack.

Say we have something like this:

```
foo: (u256, address, u256) = (42, 0x01, 26)
```

The total size of this value is 84 bytes, so it can not be stored in one stack slot. If we were to commit to storing tuple values on the stack, then we would need to map it to something like this:

```
let foo_0 := 42
let foo_1 := 0x01
let foo_2 := 26
```

This seems pretty messy to deal with... Imagine trying to pass a whole tuple into another function like this :thinking:. Using memory seems alot simpler imo because we just have to deal with one pointer:

```
let foo := alloc(84)
mstoren(foo, 42, 32)
mstoren(add(foo, 32), 0x01, 20)
mstoren(add(foo, 52), 26, 32)
```

Then looking up values in the tuple is pretty similar to looking up values in an array, the only difference being that we must know the index at compile time so that we can perform type analysis.

Reply:
Ah, that makes sense!